File: src/main/java/com/ggardet/hospital/spi/external/ExternalService.java
package com.ggardet.hospital.spi.external;

import com.ggardet.hospital.spi.ExternalInterface;
import com.ggardet.core.HospitalDeletedEvent;
import lombok.RequiredArgsConstructor;
import org.springframework.context.ApplicationEventPublisher;
import org.springframework.stereotype.Component;

import java.util.UUID;

@Component
@RequiredArgsConstructor
class ExternalService implements ExternalInterface {
    private final ApplicationEventPublisher eventPublisher;

    @Override
    public void publishDeleteHospitalEvent(final UUID hospitalId) {
        eventPublisher.publishEvent(new HospitalDeletedEvent(hospitalId));
    }
}

File: src/main/kotlin/com/ggardet/hospital/spi/external/ExternalService.kt
package com.ggardet.hospital.spi.external

import com.ggardet.hospital.spi.ExternalInterface
import com.ggardet.core.HospitalDeletedEvent
import org.springframework.context.ApplicationEventPublisher
import org.springframework.stereotype.Component
import java.util.*

@Component
internal class ExternalService(
    private val eventPublisher: ApplicationEventPublisher
) : ExternalInterface {

    override fun publishDeleteHospitalEvent(hospitalId: UUID) {
        eventPublisher.publishEvent(HospitalDeletedEvent(hospitalId))
    }
}